### PR TITLE
OMID-224 Switch default logging backend to log4j2

### PR DESCRIPTION
**Pull request by stoty**
Replace logging config based on HBase
Change logging configuration for tso-server to match the HBase setup

### CODE REVIEW

**Review by gjacoby126 — COMMENTED** [review]
Looks good, just one question

File: hbase-commit-table/pom.xml [hunk 1]
```diff
@@ -112,13 +117,32 @@
...
+            <scope>test</scope>
+        </dependency>
+        <dependency>
+            <groupId>org.apache.logging.log4j</groupId>
+            <artifactId>log4j-core</artifactId>
+            <scope>test</scope>
+        </dependency>
         <dependency>
-            <groupId>org.apache.zookeeper</groupId>
-            <artifactId>zookeeper</artifactId>
```
[the remark this answers is not in the document]
Comment (stoty) [reply hbase-commit-table/pom.xml:117]: It was never needed in that module.

TBH dependencies are stil a mess in Omid.
We SHOULD depend on ZK in _timestamp-storage_ , and _tso-server_ but  the dependency was never defined there.
We also define a ZK version properrty in the top pom that we don't use now, and have never used properly.

At some point, we should clean up the dependencies and add maven-dependency-plugin check.

Now we're effectively using whatever ZK version the HBase version we're building with is pulling in, which should be fine.

**Review by apurtell — APPROVED** [review]
lgtm
I assume you used `mvn dependency:tree` to ensure all exclusions are in place to prevent log4j1 from being pulled in transitively.

**Comment by stoty** [comment]
Yes, @apurtell , the only log4j dependencies are the test/runtime log4j2 dependencies for tests and benchmarks, examples, etc.

**Pull request closed without merging**